compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -462,10 +462,11 @@ fn report_inline_asm(
     cgcx.diag_emitter.inline_asm_error(span, msg, level, source);
 }
 
-unsafe extern "C" fn diagnostic_handler(info: &DiagnosticInfo, user: *mut c_void) {
+unsafe extern "C" fn diagnostic_handler(info: Option<&DiagnosticInfo>, user: *mut c_void) {
```

Comment:
I disagree with having to wrap all references in `Option`. You have the invariant that the reference is valid (if not NULL) either way. It is very much reasonable to expect the C side not to pass NULL, in which case the argument will never be `None`.

Reply:
ok!
I'll assume you have the same to say about the changes in `compiler/rustc_llvm/src/lib.rs`, `library/proc_macro/src/bridge/{buffer,closure}.rs`, and `src/tools/clippy/tests/ui/boxed_local.rs`, so I'll be also removing the new `Option<_>`s there as well.